cvlib/src/corner_detector.cpp

```diff
@@ -19,7 +30,81 @@ cv::Ptr<corner_detector_fast> corner_detector_fast::create()
 void corner_detector_fast::detect(cv::InputArray image, CV_OUT std::vector<cv::KeyPoint>& keypoints, cv::InputArray /*mask = cv::noArray()*/)
 {
     keypoints.clear();
+    if ((image.rows() < 7) || (image.cols() < 7))
+        return;
+    int t = 10; //threshold for fast algorithm
+    cv::Mat img_mat = image.getMat();
     // \todo implement FAST with minimal LOCs(lines of code), but keep code readable.
+
+    for (int i = 3; i < image.rows() - 3; i++)
+        for (int j = 3; j < image.cols() - 3; j++)
+        {
+            int �ompare_arr[17] = {0}; // compare result array (-1 - darker, 0 - same, 1 - lighter)
+            cv::Point pix_pos = cv::Point(j, i);
+            int Ip = img_mat.at<uint8_t>(pix_pos); // suspect pixel
+            //check pixels 1, 5, 9, 13
+            int count_l = 0, count_d = 0; //counters for lighter and darker pixels
+            for (int k = 1; k <= 13; k += 4)
+            {
+                int Ipi = img_mat.at<uint8_t>(pix_pos + fast_offsets[k]);
+                if (Ipi > Ip + t)
+                {
+                    �ompare_arr[k] = 1;
+                    count_l++;
+                }
+                else if (Ipi < Ip - t)
+                {
+                    �ompare_arr[k] = -1;
+                    count_d++;
+                }
+                else
+                    �ompare_arr[k] = 0;
+            }
+            if ((count_l >= 3) || (count_d >= 3)) 
+            {
+                // The pixels have been verified. Check the remaining pixels
+                for (int k = 1; k <= 16; k++)
+                {
+                    if ((k == 1) || (k == 5) || (k == 9) || (k == 13))
+                        continue;
+                    int Ipi = img_mat.at<uint8_t>(pix_pos + fast_offsets[k]);
+                    if (Ipi > Ip + t)
+                        �ompare_arr[k] = 1;
+                    else if (Ipi < Ip - t)
+                        �ompare_arr[k] = -1;
+                    else
+                        �ompare_arr[k] = 0;
+                }
+                if (check_count_same_in_a_row(�ompare_arr, 12))
```

Comment:
```suggestion
                if (check_count_same_in_a_row(�ompare_arr + 1, 12))
```
Врозе вот так должно быть, первый элемент же не учитывается?

Reply:
согласен